netket/operator/_kinetic.py

```diff
@@ -89,10 +96,26 @@ def logpsi_x(x):
             return logpsi(params, x)
 
         dlogpsi_x = jacrev(logpsi_x)
-
-        dp_dx2 = jnp.diag(jacfwd(dlogpsi_x)(x)[0].reshape(x.shape[0], x.shape[0]))
         dp_dx = dlogpsi_x(x)[0][0] ** 2
 
+        # multiple methods to compute the second order derivatives
+        if self._method == "parallel":
+            dp_dx2 = jnp.diag(jacfwd(dlogpsi_x)(x)[0].reshape(x.shape[0], x.shape[0]))
+        elif self._method == "sequential":
+
+            def make_hess(carry, vec):
+                value = jacfwd(dlogpsi_x, vec=vec)(x)[0]
+                return carry, value
+
+            _, dp_dx2 = jax.lax.scan(
+                make_hess, None, jnp.eye(x.shape[0], dtype=x.dtype)
+            )
+            dp_dx2 = jnp.diag(dp_dx2.reshape(x.shape[0], x.shape[0]))
+        else:
```

Comment:
We could add Feenberg as:
```
elif self._method == "Feenberg":
return 0.5*jnp.sum(inverse_mass * dp_dx**2, axis=-1)
```
although this will only be valid for open systems (wave-function goes to zero at infinity) or periodic systems where the wavefunction is forced to zero gradient at the boundaries.